source/agora/node/Config.d

```diff
@@ -467,6 +469,16 @@ public struct RegistryConfig
             if (!zone.primary.set)
                 return;
 
+            foreach (addr; zone.allow_transfer)
+            {
+                try
+                    parseAddress(addr);
+                catch (Exception)
+                    ensure(0,
+                        "allow_transfer: Field must consist from IP addresses, {} is not an IP address",
+                        addr);
+            }
```

Comment:
I think you could get away with `guessAddressType`, which doesn't `throw`.
Also if you create a simple type for it with a `fromString` / `this (string)` and throw from there, you will get the file/line and field name printed for you by the config parser.